app/controllers/devise/two_step_verification_session_controller.rb

```diff
@@ -0,0 +1,32 @@
+class Devise::TwoStepVerificationSessionController < DeviseController
+  skip_before_action :handle_two_step_verification
+  def new
+  end
+
+  def create
```

Comment:
Maybe out of scope for this particular PR, but this action really could do with a tidy up. The logic here is getting a little gnarly. :eyes: